sql/core/src/main/scala/org/apache/spark/sql/execution/ReplaceHashWithSortAgg.scala

```diff
@@ -94,9 +94,13 @@ object ReplaceHashWithSortAgg extends Rule[SparkPlan] {
   /**
    * Check if `partialAgg` to be partial aggregate of `finalAgg`.
    */
-  private def isPartialAgg(partialAgg: BaseAggregateExec, finalAgg: BaseAggregateExec): Boolean = {
+  def isPartialAgg(partialAgg: BaseAggregateExec, finalAgg: BaseAggregateExec): Boolean = {
+    // We should make sure the groupingExpressions of `partialAgg` and `finalAgg` are same
+    // to avoid final aggregate depends on the output of partial aggregate.
+    // For example: `Aggregate(Seq(Alias), Seq(Alias), ..)`.
     if (partialAgg.aggregateExpressions.forall(_.mode == Partial) &&
-        finalAgg.aggregateExpressions.forall(_.mode == Final)) {
+        finalAgg.aggregateExpressions.forall(_.mode == Final) &&
+        partialAgg.groupingExpressions == finalAgg.groupingExpressions) {
```

Comment:
this is a kind of bug before, as we construct `Aggregate(Seq(Alias), Seq(Alias), ..)` pattern in DPP